subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -102,11 +105,68 @@
  */
 @DisableCachingByDefault(because = "Abstract super-class, not to be instantiated directly")
 public abstract class AbstractTestTask extends ConventionTask implements VerificationTask, Reporting<TestTaskReports> {
+
+    /**
+     * Wraps a list of listeners to subscribe, and lazily configures an anonymous broadcaster with those listeners when requested.
+     * Instances of this class are suitable for serialization (as long as listeners are serializable as well).
+     */
+    private class BroadcastSubscriptions<T> {
+        private final Class<T> listenerClass;
+        private final List<Object> subscribedListeners = new LinkedList<Object>();
+        private transient ListenerBroadcast<T> broadcaster;
+
+        private BroadcastSubscriptions(Class<T> listenerClass) {
+            this.listenerClass = listenerClass;
+        }
+
+        @SuppressWarnings("unchecked")
+        public synchronized ListenerBroadcast<T> get() {
+            if (broadcaster == null) {
+                broadcaster = getListenerManager().createAnonymousBroadcaster(listenerClass);
+                for (Object listener: subscribedListeners) {
+                    if (listenerClass.isInstance(listener)) {
```

Comment:
💅 A transaction log structure could get rid of these execution time decisions by recording the original configuration time decision.

Reply:
I had that at first, but with lack of support for lambdas, quite more code was needed (one inner class for each combination of event vs closure/listener).

Reply:
Hm, I see.